XUnit-BrowserStack/ParallelTest.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Xunit;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Remote;
+
+namespace XUnit_BrowserStack
+{
+    public abstract class ParallelBaseTest
+    {
+        private readonly BaseFixture baseFixture;
+
+        public ParallelBaseTest(BaseFixture baseFixture)
+        {
+            this.baseFixture = baseFixture;
+        }
+
+        public void BaseTest(string platform)
+        {
+            try
+            {
+                RemoteWebDriver driver = baseFixture.GetDriver(platform, "parallel");
+                driver.Manage().Window.Maximize();
+                driver.Navigate().GoToUrl("https://bstackdemo.com/");
+                Assert.Equal("StackDemo", driver.Title);
+
+                string productOnPageText = driver.FindElement(By.XPath("//*[@id=\"1\"]/p")).Text;
+
+                driver.FindElement(By.XPath("//*[@id=\"1\"]/div[4]")).Click();
+                bool cartOpened = driver.FindElement(By.XPath("//*[@class=\"float-cart__content\"]")).Displayed;
+                Assert.True(cartOpened);
+                string productOnCartText = driver.FindElement(By.XPath("//*[@id=\"__next\"]/div/div/div[2]/div[2]/div[2]/div/div[3]/p[1]")).Text;
+                Assert.Equal(productOnCartText, productOnPageText);
```

Comment:
Is timeout required to ensure tests don't fail?

Reply:
I will just add WebDriverWait for the find elements.